src/render/tutorials/getting-started-with-docker/2_0.html.md

```diff
@@ -155,72 +199,216 @@ Add in `routingOperations` array following definition:
+
+```
+modules
+└─ hellohandler
+    ├── build.gradle.kts                                                        // gradle build script
+    └── src 
+        └── main                   
+            ├── java/com/project/example/healthcheck
+            │     └── HelloWorldHandlerFactory.java                             // our handler   
+            └── resources
+                  └── META-INF/services
+                        └── io.knotx.server.api.handler.RoutingHandlerFactory   // META-INF file used by Knot.x to find the handler   
+```
+
+Let's take a look at the three files required for our module to work.
+
+`build.gradle.kts`
+```kotlin
+plugins {
+    `java-library`
 }
 
-actions {
-  books {
-    factory = http
-    config {
-      endpointOptions {
-        path = "/books/v1/volumes?q=java"
-        domain = www.googleapis.com
-        port = 443
-        allowedRequestHeaders = ["Content-Type"]
-      }
-      webClientOptions {
-        ssl = true
-      }
+dependencies {
+
+    "io.knotx:knotx".let { v ->
+        implementation(platform("$v-dependencies:${project.property("knotx.version")}"))
+        api("$v-fragments-api:${project.property("knotx.version")}")
+        api("$v-fragments-handler-api:${project.property("knotx.version")}")
     }
-  }
-  te-hbs {
-    factory = knot
-    config {
-      address = knotx.knot.te.handlebars
-      deliveryOptions {
-        sendTimeout = 1000
-      }
+    "io.vertx:vertx".let { v ->
+        implementation("$v-web")
+        implementation("$v-web-client")
+        implementation("$v-rx-java2")
     }
+}
+```
+
+`io.knotx.server.api.handler.RoutingHandlerFactory`
+```
+io.knotx.example.hellohandler.HelloWorldHandlerFactory
+```
+
+In this file we list, line by line, all the [RoutingHandlerFactory](https://github.com/Knotx/knotx-server-http/blob/master/api/src/main/java/io/knotx/server/api/handler/RoutingHandlerFactory.java) implementations we want Knot.x to see. It has to be a canonical class name. It uses [Java ServiceLoader](https://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html).
+
+`HelloWorldHandlerFactory.java`
+```
+package io.knotx.example.hellohandler;
+
+import io.knotx.server.api.handler.RoutingHandlerFactory;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.ext.web.RoutingContext;
+
+public class HelloWorldHandlerFactory implements RoutingHandlerFactory {
+
+  @Override
+  public String getName() {
+    return "hellohandler";    // this has to be exactly the same string as in operations.conf file                                                 
+  }
+
+  @Override
+  public Handler<RoutingContext> create(Vertx vertx, JsonObject config) {
+    JsonObject jsonObject = new JsonObject()
+        .put("message", "Hello World From Knot.x!");
+    return event -> event.response().end(jsonObject.toString());
```

Comment:
Maybe make a note that the handler itself is an object implementing the `Handler<RoutingContext>` interface. It isn't  clear with a lambda (it might be `Function` as far as this example is concerned)

Reply:
I think it's explicit enough by the method declaration

Reply:
Right ➡️